src/localstorage.js

```diff
@@ -1,35 +1,52 @@
-export class LocalStorage {
-  constructor() {
-    this.store = {};
-  }
-
-  clear() {
-    this.store = {};
-  }
-
-  getItem(key) {
-    return this.store[key] || null;
-  }
-
-  setItem(key, value = '') {
-    // not mentioned in the spec, but we must always coerce to a string
-    this.store[key] = value + '';
-  }
-
-  removeItem(key) {
-    delete this.store[key];
-  }
-
-  key(index) {
-    const keys = Object.keys(this.store);
-    return keys[index] || null;
-  }
-
-  get length() {
-    return Object.keys(this.store).length;
-  }
-
-  toString() {
-    return '[object Storage]';
-  }
+export class LocalStorage  {
+    constructor() {
+        Object.defineProperty(this, "getItem", {
+            enumerable: false,
+            value: jest.fn((key) => {return this[key] || null })
+
+        });
+        Object.defineProperty(this, "setItem", {
+            enumerable: false,
+            value: jest.fn((key, val = '') => {
+                this[key] = val + '';
+            })
+        });
+        Object.defineProperty(this, "removeItem", {
+            enumerable: false,
+            value: jest.fn((key) => {
+                delete this[key];
+            })
+        });
+        Object.defineProperty(this, "clear", {
+            enumerable: false,
+            value: jest.fn(() => {
+                Object.keys(this).map(key => delete this[key]);
```

Comment:
I'm guessing you need to do this now instead of `this.store = {}` because you can't set this object to `{}`, right?

Reply:
correct